### PR TITLE
fix: Time based procedure phase switch (refs DPLAN-16177)

### PR DESCRIPTION
**Pull request by mussbach**
DPLAN-16177

## Summary
- Fix time-based procedure phase switching functionality
- Resolve issues with automated phase transitions based on scheduled dates
- Changed the interface from time picker to text field to avoid complex bugs and gain flexibility

## Test plan
- [ ] Test automated phase switching with configured dates
- [ ] Verify manual phase switching still works correctly
- [ ] Test edge cases with invalid/past dates

## Changes
- Updated procedure phase switching logic
- Enhanced date validation and scheduling
- Improved error handling for phase transitions

## Related Ticket
DPLAN-16177

### CODE REVIEW

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 1]
```diff
@@ -39,23 +39,50 @@
...
+                required
+              />
+              <!-- Hidden input with combined datetime for backend -->
+              <input
+                type="hidden"
+                :name="switchDateId"
+                :value="switchDate"
+              />
+            </div><!--
+         --><div class="layout__item u-1-of-3 u-pl-0_5">
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:68]:
```suggestion
         --><div class="layout__item w-1/3 pl-2">
```

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 2]
```diff
@@ -249,34 +287,133 @@ export default {
...
   methods: {
+    // Helper function to convert German date format (DD.MM.YYYY) to ISO format (YYYY-MM-DD)
+    convertGermanToIsoDate (germanDate) {
+      if (!germanDate) {
+        return ''
+      }
+
+      if (germanDate.includes('.')) {
+        const [day, month, year] = germanDate.split('.')
+        return `${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:316]:
```suggestion
        const [day, month, year] = germanDate.split('.')
        
        return `${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`
```

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 3]
```diff
@@ -249,34 +287,133 @@ export default {
...
+
+      return germanDate // Already in ISO format
+    },
+
+    // Helper function to convert date object to German format (DD.MM.YYYY)
+    formatDateToGerman (date) {
+      const day = date.getDate().toString().padStart(2, '0')
+      const month = (date.getMonth() + 1).toString().padStart(2, '0')
+      const year = date.getFullYear()
+      return `${day}.${month}.${year}`
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:327]:
```suggestion
      const year = date.getFullYear()
      
      return `${day}.${month}.${year}`
```

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 4]
```diff
@@ -249,34 +287,133 @@ export default {
...
     setDesignatedDates () {
       if (this.initSwitchDate) {
         this.autoSwitchPhase = true
+        const date = new Date(this.initSwitchDate)
+
+        this.switchDateOnly = this.formatDateToGerman(date)
+        this.switchTime = date.toTimeString().substring(0, 5)
+        this.switchDate = this.initSwitchDate
+      } else {
+        this.switchDateOnly = this.formatDateToGerman(new Date())
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:405]:
```suggestion
        this.switchDateOnly = this.minSwitchDate
```

File: client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue [hunk 5]
```diff
@@ -249,34 +287,133 @@ export default {
...
+        this.switchTime = '00:00'
       }
+
       this.switchDateMax = this.endDate
-      this.switchDate = this.initSwitchDate
-      this.startDate = formatDate(this.initSwitchDate)
+
+      if (this.switchDate) {
+        const date = new Date(this.switchDate)
+        if (!isNaN(date.getTime())) {
```
Comment (hwiem) [inline client/js/components/procedure/basicSettings/AutoSwitchProcedurePhaseForm.vue:413]:
```suggestion
        const date = new Date(this.switchDate)
        
        if (!isNaN(date.getTime())) {
```

**Review by breierDemos — APPROVED** [review]
lgtm
